mycore-base/src/main/java/org/mycore/common/content/MCRContent.java

```diff
@@ -260,11 +260,20 @@ protected MCRDataURLEncoding getDataURLEncoding() throws IOException {
 
     /**
      * Parses content, assuming it is XML, and returns the parsed document.
-     * 
+     *
+     * @return the XML document parsed from content
+     */
+    public final Document asXML() throws JDOMException, IOException, SAXException {
```

Comment:
I fail to see the mentioned API change.

Actually, there is another API change, because some child classes had deviating lists of thrown exception classes. That's why there are changes to, say, MCRMailer.

Reply:
Why not make asXML() not final and just overwrite it in the Classes which have less Exceptions?

Reply:
Because  `asXML()` is now only a convenience method that provides a default parameter for the actual method `asXML(boolean)`.

As a rule, I make all convenience methods final. This removes a common and subtle pitfall.

Making convenience methods final enforces that subclasses that wish to overwrite the functionality have to override the actual method. Otherwise it would be easy to accidentally overwrite the convenience method and create inconsistent behavior.

Reply:
Another question is whether clients of `asXML(boolean)` and `asXML(boolean)` actually care about the exact exception class or if it would suffice for them to know, that exceptions are to be expected (i.e. replace with `throws Exception`).